autogen

```diff
@@ -385,8 +395,12 @@ function autogenForFile() {
       ;;
 
     test_*.py | *_test.py)
-      echo "#!/usr/bin/python"
-      echo "#"
+
+
+      if [ ${ADD_PYTHON_CODE} -eq 1 ]; then
+        echo "#!/usr/bin/python"
+        echo "#"
```

Comment:
This line should be part of the license block; otherwise, let's just add blank line here?

Reply:
Not sure what you mean, the "#!/usr/bin/python" line is not a part of license... I am probably not understanding you, could you please elaborate

Reply:
Sorry, I misread the code, so please ignore my earlier comment.

Given this, though, are you suggesting that `#!/usr/bin/python` is also extra if `--no-code` is specified? Is the idea that non-root `*.py` files don't need a `#!/usr/bin/python` line at the top, just the license block?

What about the use case of a root Python file, where you want the `#!/usr/bin/python` line so that it's executable, but without the boilerplate code gen?

Reply:
Maybe the `#!/usr/bin/python` (or any other line) should only be eliminated if another flag, e.g., `--no-runline` is specified?

I am just suggesting that `--no-code` only removes the code _below_ the license block, and everything above the license block stays as-is. Let's discuss removing the runline in a separate issue / PR. Thoughts?

Reply:
My usage of autogen has largely been driven by a need to add OSS licenses to a large project . I found it very unusual that the tool is also trying to add actual code. ( I consider the #!/usr/bin/python code :-) ) 

So I tell the tool "go to these files, and add license in-place" and the tool does this... but also oh by the way let me add a #!/usr/bin/python line and unit test code -- which was already present in some cases and totally broke the build in others :-1:  

Is the idea that non-root *.py files don't need a #!/usr/bin/python line at the top, just the license block? - Yes, exactly , dont mess with this because as an engineer I will know what to do, in other words, just do what I want.

I realize that there may be other cases and yes,--no-runline makes sense, I ll try to add.

Reply:
I've been thinking about this a bit and realized that if there's already a `#!/usr/bin/python` at the top of the file, then we don't have a good solution here: either we add a second `#!/usr/bin/python` or if we provide the `--no-runline` then the license header will be at the top, above the existing runline, which is still invalid.

How would you fix existing files with a runline to just add a license block, but _after_ the runline? Not suggesting you should extend your current PR to handle this, just a question.

Reply:
I belived in SOLID principles - the "S" applies here, single responsibility . This tool is designed to help adding boilerplate content to files, it is really not responsible for fixing content that is already there.  So... if the file includes ! line... there must have been a reason to include it, this tool should not do more than it is designed to do...